### ISSUE TITLE
Recursive middlewares and sub-applications?

### ISSUE THREAD

**Issue by eudoxia0** [fukamachi/clack#67]
When I wrote clack-errors I ran into the problem of creating nested or recursive middlewares: The error-catching middleware displays an error page which includes CSS, but there was no way to include a static middleware to dispatch that CSS 'into' the error-catching middleware.

This limitation can show up in larger projects. For example, imagine someone writes an admin page app for Clack applications. I know Plack, which Clack is based on, includes a way to "mount" a whole Plack application, setting to root to something (ie, all requests to `/admin/index` get changed to `/index` before being sent to the Admin page sub-app), and I was wondering if something similar existed or was being planned for Clack.

Basically, what I want to do is write a Clack application with its own routes and middlewares, that will later be included as a module in another application.
